src/cargo/core/compiler/context/mod.rs

```diff
@@ -88,13 +92,19 @@ pub struct Context<'a, 'cfg: 'a> {
     pub links: Links<'a>,
     pub used_in_plugin: HashSet<Unit<'a>>,
     pub jobserver: Client,
+    pub profiles: &'a Profiles,
+    /// This is a workaround to carry the extra compiler args given on the
+    /// command-line for `cargo rustc` and `cargo rustdoc`.  These commands
+    /// only support one target, but we don't want the args passed to any
+    /// dependencies.
+    pub extra_compiler_args: HashMap<Unit<'a>, Vec<String>>,
```

Comment:
Perhaps we can use `Option<(Unit, Vec<String>)>` instead? Let's also document that this is overloaded in the sense that this can be arguments for different tools, `rustc` or `rustdoc`, depending on the command.